blocks/api/factory.js

```diff
@@ -17,3 +23,32 @@ export function createBlock( blockType, attributes = {} ) {
 		attributes
 	};
 }
+
+/**
+ * Switch Block Type and returns the updated block
+ *
+ * @param  {Object} block      Block object
+ * @param  {string} blockType  BlockType
+ * @return {Object?}           Block object
+ */
+export function switchToBlockType( block, blockType ) {
+	const destinationSettings = getBlockSettings( blockType );
+	const sourceSettings = getBlockSettings( block.blockType );
+	const transformationsFrom = get( destinationSettings, 'transforms.from', [] );
+	const transformationsTo = get( sourceSettings, 'transforms.to', [] );
+
+	// Find the from transformation
+	const transformation =
+		transformationsFrom.find( t => t.blocks.indexOf( block.blockType ) !== -1 ) ||
+		transformationsTo.find( t => t.blocks.indexOf( blockType ) !== -1 );
```

Comment:
I think the transformation `to` should come first here.  This way, if two blocks define conflicting transformations (for example, for a list block, "list `to` text" and "text `from` list"), then the current block gets the first shot at the transformation.  What do you think?